python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -1407,3 +1477,496 @@ def _compute_model_params(self, wcsf):
+    ):
+        """Match the input catalog of isolated stars with the reference catalog
+        and transform the combined isolated star sources and reference source
+        into the format needed for gbdes.
+
+        Parameters
+        ----------
+        isolatedStarSourceRefs : `list`
+            List of `DeferredDatasetHandle`s pointing to isolated star sources.
+        isolatedStarCatalogRefs: `list`
+            List of `DeferredDatasetHandle`s pointing to isolated star
+            catalogs.
+        extensionInfo : `lsst.pipe.base.Struct`
+            Struct containing properties for each extension.
+        refObjects : `dict`
+            Dictionary of dictionaries containing the position and error
+            information of reference objects.
+
+        Returns
+        -------
+        associations : `lsst.pipe.base.Struct`
+            Struct containing the source associations.
+        sourceDict : `dict`
+            Dictionary containing the source centroids for each visit.
+        """
+        sequences = []
+        extensions = []
+        object_indices = []
+
+        sourceColumns = ["x", "y", "xErr", "yErr", "ixx", "ixy", "iyy", "obj_index", "visit", "detector"]
+        catalogColumns = ["ra", "dec"]
+
+        sourceDict = dict([(visit, {}) for visit in np.unique(extensionInfo.visit)])
+        for visit, detector in zip(extensionInfo.visit, extensionInfo.detector):
+            sourceDict[visit][detector] = {"x": [], "y": [], "xCov": [], "yCov": [], "xyCov": []}
+
+        for isolatedStarCatalogRef, isolatedStarSourceRef in zip(
+            isolatedStarCatalogRefs, isolatedStarSourceRefs
+        ):
+            isolatedStarCatalog = isolatedStarCatalogRef.get(parameters={"columns": catalogColumns})
+            isolatedStarSources = isolatedStarSourceRef.get(parameters={"columns": sourceColumns})
+            if len(isolatedStarCatalog) == 0:
+                continue
+
+            # Match the reference stars to the existing isolated stars, then
+            # insert the reference stars into the isolated star sources.
+            allVisits = np.copy(isolatedStarSources["visit"])
+            allDetectors = np.copy(isolatedStarSources["detector"])
+            allObjectIndices = np.copy(isolatedStarSources["obj_index"])
+            issIndices = np.copy(isolatedStarSources.index)
+            for f, regionRefObjects in refObjects.items():
+                # Use the same matching technique that is done in
+                # isolatedStarAssociation and fgcmBuildFromIsolatedStars:
+                with Matcher(
+                    isolatedStarCatalog["ra"].to_numpy(), isolatedStarCatalog["dec"].to_numpy()
+                ) as matcher:
+                    idx, i1, i2, d = matcher.query_radius(
+                        np.array(regionRefObjects["ra"]),
+                        np.array(regionRefObjects["dec"]),
+                        self.config.matchRadius / 3600.0,
```

Comment:
No this is associating a reference catalog with the isolated star catalog, which comes from isolating the source catalogs that come out of single frame processing, so an `arcsec` is generous but not crazy high. If an object from the reference catalog gets associated with multiple stars from the isolated star catalog, those matches will get dropped from the fit (internally in `gbdes`). This would only happen if the reference catalog goes significantly deeper than the science images.